doc/src/explanation/solve_output.rst

```diff
@@ -0,0 +1,142 @@
+    all linear in the symbols given. Note: such a system is automatically generated
+    for a single equation (not passed as a list) if there is an
+    undetermined-coefficients solution for the symbols specified. If this is not
+    what was intended, then pass the expression in a list.
+
+    >>> solve([x + y - 2, x - y + 2], x, y)
+    {x: 0, y: 2}
+    >>> eq = a*x - 2*x + b - 5
+    >>> solve(eq, {a, b})  # undetermined coefficients
+    {a: 2, b: 5}
+    >>> solve([eq], {a, b})  # algebraic
+    {a: -b/x + (2*x + 5)/x}
+
+List of Tuples
+--------------
+
+    Each tuple in the list gives a solution for the symbols in the order
+    they were given. This format is used when a) a list of equations contains
+    at least one nonlinear equation or b) a list of symbols is given in a
+    well defined order. (This is also the format for the tuples in the set
+    returned when using the flag ``set=True``.)
+
+    >>> solve(x - 1, x, y)  # more than one symbol
+    [(1, y)]
+    >>> solve([x**2], x)  # list with nonlinear equation
+    [(0,)]
+    >>> solve([x**2 - 1], x)
+    [(-1,), (1,)]
+    >>> solve([x**2 - y, x - 3], x, y)  # nonlinear and multiple symbols
+    [(3, 9)]
+
+List of Dictionaries
+--------------------
+
+    The list of dictionaries is returned when the expression was not
+    univariate or there was a nonlinear expression in a list *and* the order of
+    symbols would otherwise be ambiguous because a) no symbols were passed or
+    b) the symbols were passed as a set. (This is also the format selected with
+    ``dict=True``.)
+
+    >>> solve(x - y)
+    [{x: y}]
+    >>> solve([exp(x) - 1, x*(x - 1)])
+    [{x: 0}]
+    >>> system = [x + y - z, x**2 - y + z, exp(z) + 1/x + 1/y - 2]
+    >>> sol = solve(system[:2]); sol
+    [{x: -1, y: z + 1}, {x: 0, y: z}]
+
+    The dictionaries only contain values that are distinct from the keys.
+    In the last example above, there is no key for ``z`` in the dictionary
+    since only *two* of the three equations were insufficient to determine its
+    value. These solutions can be used to eliminate those variables from the
+    third equation, however, to give a relationship in a single variable that
+    can be solved (perhaps numerically) to obtain a full solution with the
+    advantage of only needing to guess a single value instead of three.
+
+        >>> from sympy import nsolve
+        >>> [system[-1].subs(s) for s in sol]
+        [exp(z) - 3 + 1/(z + 1), exp(z) + zoo + 1/z]
+        >>> z_eq = _[0]
+        >>> zsol = nsolve(z_eq, 1); zsol
+        0.906425478894557
+        >>> sol0 = {k: v.subs(z, zsol) for k, v in sol[0].items()}
+        >>> sol0[z] = zsol; sol0
+        {x: -1, y: 1.90642547889456, z: 0.906425478894557}
+
+Boolean or Relational
+---------------------
+
+    A boolean expression is returned when a relational expression other
+    than an Equality is given as an expression to solve. A single Equality
```

Comment:
Suggest linking to Equality [sympy.core.relational.Equality](https://docs.sympy.org/dev/modules/core.html?highlight=equality#sympy.core.relational.Equality) on first use